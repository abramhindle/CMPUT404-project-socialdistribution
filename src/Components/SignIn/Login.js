import React, { useEffect, useState } from "react";
import { Button, Input, Panel, InputGroup, Message, useToaster } from "rsuite";
import EyeIcon from "@rsuite/icons/legacy/Eye";
import EyeSlashIcon from "@rsuite/icons/legacy/EyeSlash";
import axios from "axios";
import {
	getCsrfToken,
	setCurrentUser,
	setLoggedIn,
	unsetCurrentUser,
	setCreds,
} from "../utils/auth";
import { useNavigate } from "react-router-dom";

function LOGIN() {
	const [username, set_username] = useState("");
	const [password, set_password] = useState("");
	const [visible, setVisible] = React.useState(false);
	let toaster = useToaster();
	let navigate = useNavigate();

	useEffect(() => {
		if (localStorage.getItem("loggedIn")) {
			navigate("/");
		} else {
			unsetCurrentUser();
			getCsrfToken();
		}
	}, []);

	const handleChange = () => {
		setVisible(!visible);
	};

	const notifyFailedPost = (error) => {
		toaster.push(<Message type="error">{error}</Message>, {
			placement: "topEnd",
			duration: 5000,
		});
	};

	function delay(time) {
		return new Promise((resolve) => setTimeout(resolve, time));
	}

	delay(1000).then(() => console.log("ran after 1 second1 passed"));

	async function handleLoginClick() {
		var params = {
			username: username,
			password: password,
		};
		await getCsrfToken();
		const token = localStorage.getItem("token");

		let reqInstance = axios.create({
			headers: { "X-CSRFToken": token },
			baseURL: `http://127.0.0.1:8000/`,
		});
		reqInstance({ method: "post", url: "dlogin", data: params })
			.then(async (res) => {
				getCsrfToken();
				setLoggedIn(true);
				setCreds(params);
				await setCurrentUser(res.data).then(navigate("/"));
			})
			.catch((err) => notifyFailedPost(err.response.data));
	}

	return (
		<div>
			<Panel header="Login" shaded style={{ textAlign: "center" }}>
				<Input
					onChange={(e) => set_username(e)}
					value={username}
					placeholder="username"
					style={{ width: "70%", margin: "auto" }}
				/>
				<InputGroup
					inside
					style={{ width: "70%", margin: "auto", marginTop: "10px" }}
				>
					<Input
						onChange={(e) => set_password(e)}
						value={password}
						placeholder="password"
						type={visible ? "text" : "password"}
					/>
					<InputGroup.Button onClick={handleChange}>
						{visible ? <EyeIcon /> : <EyeSlashIcon />}
					</InputGroup.Button>
				</InputGroup>
				<Button
					apperance="Primary"
					block
					style={{ width: "70%", margin: "auto", marginTop: 10 }}
					onClick={handleLoginClick}
				>
					Login
				</Button>
			</Panel>
		</div>
	);
}

export default LOGIN;
