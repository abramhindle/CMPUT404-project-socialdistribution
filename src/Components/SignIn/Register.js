import React, { useState } from "react";
import { Button, Input, Panel, InputGroup, Message, useToaster } from "rsuite";
import EyeIcon from "@rsuite/icons/legacy/Eye";
import EyeSlashIcon from "@rsuite/icons/legacy/EyeSlash";
import { reqInstance } from "../utils/axios";
import { getCsrfToken } from "../utils/auth";

function REGISTER() {
	const [username, set_username] = useState("");
	const [password, set_password] = useState("");
	const [fullname, set_fullname] = useState("");
	const [email, set_email] = useState("");
	const [visible, setVisible] = React.useState(false);
	let toaster = useToaster();

	const handleChange = () => {
		setVisible(!visible);
	};

	const notifySuccessPost = (message) => {
		toaster.push(<Message type="success">{message}</Message>, {
			placement: "topEnd",
			duration: 5000,
		});
	};

	const notifyFailedPost = (error) => {
		toaster.push(<Message type="error">{error}</Message>, {
			placement: "topEnd",
			duration: 5000,
		});
	};

	const handleSubmitClick = () => {
		var params = {
			username: username,
			password: password,
			fullname: fullname,
			email: email,
		};
		getCsrfToken();

		reqInstance({ method: "post", url: "dregister", data: params })
			.then((res) => {
				notifySuccessPost("Registration Successful");
			})
			.catch((err) => notifyFailedPost(err));
	};

	return (
		<div>
			<Panel
				header="Register"
				shaded
				style={{ margin: "auto", textAlign: "center" }}
			>
				<Input
					onChange={(e) => set_email(e)}
					placeholder="Email"
					style={{ width: "70%", margin: "auto" }}
				/>
				<Input
					onChange={(e) => set_fullname(e)}
					placeholder="Full Name"
					style={{ width: "70%", margin: "auto", marginTop: "10px" }}
				/>
				<Input
					onChange={(e) => set_username(e)}
					placeholder="Display Name"
					style={{ width: "70%", margin: "auto", marginTop: "10px" }}
				/>
				<InputGroup
					inside
					style={{ width: "70%", margin: "auto", marginTop: "10px" }}
				>
					<Input
						onChange={(e) => set_password(e)}
						placeholder="password"
						type={visible ? "text" : "password"}
					/>
					<InputGroup.Button onClick={handleChange}>
						{visible ? <EyeIcon /> : <EyeSlashIcon />}
					</InputGroup.Button>
				</InputGroup>
				<Button
					apperance="Primary"
					block
					style={{ width: "70%", margin: "auto", marginTop: "10px" }}
					onClick={handleSubmitClick}
				>
					Register
				</Button>
			</Panel>
		</div>
	);
}

export default REGISTER;
