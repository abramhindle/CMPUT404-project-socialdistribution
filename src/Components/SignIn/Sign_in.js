import React, { useState } from "react";
import { Button, ButtonToolbar, ButtonGroup } from "rsuite";
// Component Imports
import LOGIN from "./Login";
import REGISTER from "./Register";

function SIGN_IN() {
	const [Login, setLogin] = React.useState(true);
	const [appearance, setAppearance] = React.useState({
		login: "primary",
		register: "ghost",
	});

	const handleLoginBtnClick = () => {
		setLogin(true);
		setAppearance({ login: "primary", register: "ghost" });
	};

	const handleRegisterBtnClick = () => {
		setLogin(false);
		setAppearance({ login: "ghost", register: "primary" });
	};

	return (
		<div style={{ width: "40%", margin: "auto" }}>
			<ButtonToolbar style={{ width: "100%" }}>
				<ButtonGroup justified>
					<Button
						style={{ textAlign: "center" }}
						appearance={appearance["login"]}
						onClick={handleLoginBtnClick}
					>
						Login
					</Button>
					<Button
						style={{ textAlign: "center" }}
						appearance={appearance["register"]}
						onClick={handleRegisterBtnClick}
					>
						Register
					</Button>
				</ButtonGroup>
			</ButtonToolbar>
			{Login ? <LOGIN /> : <REGISTER />}
		</div>
	);
}

export default SIGN_IN;
