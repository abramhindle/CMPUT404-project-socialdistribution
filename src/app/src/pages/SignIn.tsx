import styled from 'styled-components';
import TextField, { TextFieldProps } from '@mui/material/TextField';
import Button, { ButtonProps } from '@mui/material/Button';
import { makeStyles, styled as Styled } from '@mui/material/styles';
import { useState } from 'react';

const ColorButton = Styled(Button)<ButtonProps>(({ theme }) => ({
  color: theme.palette.getContrastText('#e6c9a8'),
  padding: '10px',
  marginTop: '2%',
  backgroundColor: '#e6c9a8',
  '&:hover': {
    backgroundColor: '#D4AF85',
  },
}));

const TextFieldCustom = Styled(TextField)<TextFieldProps>(({ theme }) => ({
  color: theme.palette.getContrastText('#fbf8f0'),
  height: '50px',
  padding: '10px',
  '& .Mui-focused': {
    color: 'black',
  },
  backgroundColor: 'white',
  '&:hover': {
    backgroundColor: '#fbf8f0',
  },
}));

export default function SignIn() {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const handleClick = () => {
    console.log(username, password);
  };
  const onChangeUsername = (event: any) => {
    setUsername(event?.target?.value);
  };
  const onChangePassword = (event: any) => {
    setPassword(event?.target?.value);
  };
  return (
    <>
      <TextFieldCustom
        id="standard-basic"
        color="primary"
        label="Username"
        variant="standard"
        onChange={onChangeUsername}
      />
      <TextFieldCustom
        id="standard-basic"
        label="Password"
        variant="standard"
        onChange={onChangePassword}
      />
      <ColorButton variant="contained" onClick={handleClick}>
        Sign in
      </ColorButton>
    </>
  );
}
