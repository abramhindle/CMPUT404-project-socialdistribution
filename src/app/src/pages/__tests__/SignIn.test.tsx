import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import SignIn from '../SignIn';

let container: any = null;
//issues with container type and its methods

beforeEach(() => {
  // setup a DOM element as a render target
  container = document?.createElement('div');
  document?.body?.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container?.remove();
  container = null;
});

describe('SignUp suite', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<SignIn setCurrentUser={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
