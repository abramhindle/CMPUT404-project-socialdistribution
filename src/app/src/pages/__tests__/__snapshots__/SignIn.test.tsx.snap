// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp suite Renders correctly 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root css-lxpoe1-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Email
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root css-lxpoe1-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Password
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="current-password"
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
      />
    </div>
  </div>,
  <button
    className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-abq3u9-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Sign in
  </button>,
]
`;
